Apsitvarkom.ModelActions/Mapping/CleaningEventProfile.cs

```diff
@@ -21,14 +22,18 @@ private void MapRequests()
             .ForMember(x => x.Id, opt => opt.MapFrom(x => x.Id));
         CreateMap<CleaningEventCreateRequest, CleaningEvent>()
             .ForMember(x => x.Id, opt => opt.Ignore())
-            .ForMember(x => x.PollutedLocation, opt => opt.Ignore());
+            .ForMember(x => x.IsFinalized, opt => opt.Ignore());
         CreateMap<CleaningEventUpdateRequest, CleaningEvent>()
             .ForMember(x => x.PollutedLocationId, opt => opt.Ignore())
-            .ForMember(x => x.PollutedLocation, opt => opt.Ignore());
+            .ForMember(x => x.IsFinalized, opt => opt.Ignore());
     }
 
     private void MapResponses()
     {
-        CreateMap<CleaningEvent, CleaningEventResponse>();
+        CreateMap<CleaningEvent, CleaningEventResponse>()
+            .ForMember(x => x.Status, opt => 
+                opt.MapFrom(x => x.IsFinalized ? CleaningEventResponse.State.Finalized 
+                                                                        : x.StartTime > DateTime.UtcNow ? CleaningEventResponse.State.Foreseen 
+                                                                                                        : CleaningEventResponse.State.Finished));
```

Comment:
Pattern matching?

Reply:
DateTime.UtcNow is not a constant and we can have a statement body because this is a lambda expression. Will slightly adjust the indentation to make it more readable this way instead.